docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+Implement support for coordinate systems. Add an interface for `SkyCoord`, by
+introducing a `SpatialModel.skycood` attribute.
+
+.. code-block::
+
+    from astropy.coordinates import SkyCoord
+    from gammapy.image.models import PointSource
+
+    # instanciation using a SkyCoord object
+    point_source = PointSource(skycood=SkyCoord(0, 0, frame="galactic", unit="deg"))
+    # or
+    point_source = PointSource(lon="0 deg", lat="0 deg", frame="galactic")
+
+    skycoord = point_source.skycoord
+    assert isinstance(skycoord, SkyCoord)
+
+    # evaluation of models using sky coordinates
+    coords = SkyCoord([0, 1], [0.5, 0.5], frame="galactic", unit="deg")
+    values = point_source(coords)
+
+    # maybe override evaluation for efficient evaluation for fitting
+    values = point_source(lon, lat)
+
+
+Implement default parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For spectral models Gammapy has defined default parameters, which simplifes the
+interactive usage of the models. From this experience we propose to introduce
+default parameters for `SpatialModel` as well as `SourceModel`.
+
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+    from gammapy.cube.models import SourceModels
+
+    shell = Shell()
+    print(shell)
+
+
+Implement evaluation region specifications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For efficient evaluation of model components, we propose that `SpatialModel`
+classes declare a circular region `.evaluation_radius`, they are evaluated on.
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+
+    shell = Shell()
+    print(shell.evaluation_radius)
+
+The `.evaluation_radius` should be a property of the model, that is computed
+depending of the model parameter, that determines the spatial extension of the
+model. The handling of `PointSource` and additional evaluation margins, which depend
+on the pixel size, is done in the step, where the model component is evaluated.
+
+
+Expose model parameters as attributes
+-------------------------------------
+To simplify the interactive handling of model instances, model parameters should
+be exposed as attribues on the model, so that the following works:
+
+.. code-block::
+
+    from gammapy.spectrum.models import Powerlaw
+
+    pwl = Powerlaw()
+
+    pwl.amplitude.value = "1e-12 cm-2 s-1 TeV-1"
+
+    # instead of
+
+    pwl.parameters["amplitude"].value = "1e-12 cm-2 s-1 TeV-1"
+
+The attributes should be auto-genrated in the `Model` base class, based on the
+list of parameters defined by the model.
+
+Introduce sub-module for models
+-------------------------------
+
+We propose to introduce a `gammapy.models` or `gammapy.modeling` sub-module, which
+bundles all the model classes in single namespace.
+
+.. code-block::
+
+    from gammapy.models import Powerlaw, SpatialGaussian, SourceModel
+
+    # or consistently with astropy
+
+    from gammapy.modeling import Powerlaw, SpatialGaussian, SourceModel
+
+
+List of Pull Requests
```

Comment:
Suggest to number the pull requests, i.e. make a bullet list like this:
```
#. one
#. two
```
There's 10+, so discussion is easier if there's numbers, and I also think it's good to see how many there are.

Also, I think several should be split in two PRs. In my experience it's more efficient to have small PRs.

I also think it would be helpful to suggest a development cycle (v0.10, v0.11, v0.12) where this could go.
I.e. basically this PIG very clearly outlines the work to be done, so that people can just pick up a task.
Only if unclear / needed could the lead devs then make an issue for each task that describes it in greater detail, otherwise someone would just start a PR (can be with just a few issues) and discussion could go there directly. This would guarantee that discussion is concrete and useful, and avoid very much time spent on planning and disucssion ahead, without it being clear if / who will do the work.

I would suggest to put the three PRs with "rename" at the end. They are very disturbing / break every user script. And really aren't high priority, not needed to make progress. They can also be in the middle like you have it now, but then they should probably be all in one given release (e.g. v0.11), instead of being spread across many releases, no?